File: docs/snippets/all/concepts/file_sink.cpp
// Create and set a file sink.

#include <rerun.hpp>

int main(int argc, char* argv[]) {
    const auto rec = rerun::RecordingStream("rerun_example_file_sink");

    rec.set_sinks(rerun::FileSink{"recording.rrd"}).exit_on_failure();
}

File: docs/snippets/all/concepts/file_sink.rs
//! Create and set a file sink.

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = rerun::RecordingStreamBuilder::new("rerun_example_file_sink").buffered()?;

    rec.set_sink(Box::new(rerun::sink::FileSink::new("recording.rrd")?));

    Ok(())
}
